### PR TITLE
Fix invalid html

### PR DESCRIPTION
**Pull request by teceler**

### CODE REVIEW

File: app/views/users/new.haml [hunk 1]
```diff
@@ -4,7 +4,7 @@
       %tr
         %th{colspan: 2} Sign Up
     %tbody
-      %tr#username
+      %tr#username-row
```
Comment (Marri) [inline app/views/users/new.haml]: I am reasonably confident despite not having checked that this will break the user validation javascript

Comment (Throne3d) [reply app/views/users/new.haml]: Can we call these like `#signup-username`, `#signup-password`, `#signup-secret` btw? That way we describe their uniqueness better. Just before we go changing the javascript.

Comment (teceler) [reply app/views/users/new.haml]: Can do

File: app/views/posts/_list_item.haml [hunk 2]
```diff
@@ -53,7 +53,7 @@
     by
     = user_mem_link(post.last_user_id, post.last_user_name, post.last_user_deleted?)
   - if local_assigns[:check_box_name]
-    %td.post-check-box{class: klass}= check_box_tag check_box_name, post.id, false, class: 'checkbox'
+    %td.post-check-box{class: klass}= check_box_tag check_box_name, post.id, false, class: 'checkbox', id: "#{check_box_name.delete('[]')}-#{post.id}"
```
Comment (Throne3d) [inline app/views/posts/_list_item.haml]: Do we use this ID? It's a little weird that we're naming it, for example, `marked_ids_9`, if we don't actually need an ID, so maybe it should have `id: nil`?

File: app/views/messages/index.haml [hunk 3]
```diff
@@ -28,7 +28,7 @@
               - else
                 = user_link(message.recipient)
             %td{class: klass}= pretty_time(message.last_in_thread.created_at)
-            %td{class: klass}= check_box_tag :"marked_ids[]", message.id, false, class: 'select-all-box'
+            %td{class: klass}= check_box_tag :"marked_ids[]", message.id, false, class: 'select-all-box', id: "marked_ids_#{message.id}"
```
Comment (Throne3d) [inline app/views/messages/index.haml]: Again wondering if we need IDs here

Comment (Throne3d) [reply app/views/messages/index.haml]: Applies to the other checkboxes too, actually.

File: app/views/password_resets/new.haml [hunk 4]
```diff
@@ -4,7 +4,7 @@
       %th.centered{colspan: 2} Request Password Reset
     %tr
       %th.sub Username
-      %td.even= text_field_tag :username, params[:username], placeholder: "Username", class: 'text'
+      %td.even= text_field_tag :username, params[:username], placeholder: "Username", class: 'text', id: 'user_username'
     %tr
       %th.sub Email
       %td.odd= text_field_tag :email, params[:email], placeholder: "Email address", class: 'text'
```
Comment (Throne3d) [inline app/views/password_resets/new.haml]: For consistency, give this `id: 'user_email'`?

File: app/views/users/new.haml (hunk 1 above)
Comment (Throne3d) [reply app/views/users/new.haml]: Yep, user validation javascript will need tweaking: <https://github.com/Marri/glowfic/blob/9692cda6d8ecc3b53f1d7058d57d5cbc5714e812/app/assets/javascripts/users/new.js#L3>

Specifically, `#username`, `#email`, `#password`, and also most uses of `addAlertAfter`.

**Comment by Throne3d** [comment]
Needs the javascript change before further review.

File: app/views/galleries/_single.haml [hunk 5]
```diff
@@ -47,7 +47,7 @@
                 %br
                 %span.icon-keyword= keyword
               - if is_owner && !skip_forms
-                .select-button= check_box_tag :"marked_ids[]", icon_id
+                .select-button= check_box_tag :"marked_ids[]", icon_id, true, id: nil
```
Comment (Throne3d) [inline app/views/galleries/_single.haml]: This should default to false – currently defaults to making all icon check boxes ticked on gallery#show

**Comment by Throne3d** [comment]
- gallery select all buttons still work on add_existing
- character replace post_id checkboxes still work
- messages select all button still works
- "no icon" displays fine
- marking icons in a gallery works fine (for deleting), but see comment
- submits `return_tag` when removing icons from a gallery in a tag#show?view=galleries expandable
- hides posts fine (`marked_ids` there)
- switching per_page works
- mobile pagination shows `&raquo;` fine (it did before, too, though, which was weird)
- username, email submission on password reset is fine
- javascript on signup form works, submitting sends params

**Pull request merged**